src/papi-client.ts

```diff
@@ -0,0 +1,96 @@
+import {EventMessage} from "./models";
+import * as Promise from "bluebird";
+import {get, post, put, Response} from "superagent";
+
+export type BatchId = number;
+
+export interface PapiCredentials {
+    login: string;
+    password: string;
+}
+
+const DEFAUT_DELAY_MS = 1000;
+
+export class PapiClient {
+
+    public currentBatchId: BatchId;
+
+    constructor(
+        private apiUrl: string,
+        private domainUUID: string,
+        private credentials: PapiCredentials
+    ) {}
+
+    public startBatch(): Promise<Response> {
+        if (this.currentBatchId) {
+            throw new Error("The following batch is already started: " + this.currentBatchId);
+        }
+
+        return Promise.fromCallback(callback => {
+                this.auth(post(this.papiUrl("/batches/"))).end(callback);
```

Comment:
I already tried to make that better but without success. Now I've a better understanding of TS so I finally found a solution. Done.